cdk/lib/cdk-stack.ts

```diff
@@ -55,6 +57,86 @@ export class DeploymentService extends cdk.Stack {
       }
     );
 
+    const getProductsListFunction = new lambda.Function(
+      this,
+      "getProductsListFunction",
+      {
+        runtime: lambda.Runtime.NODEJS_18_X,
+        handler: "getProductsList.handler",
+        code: lambda.Code.fromAsset("lambda"),
+      }
+    );
+    const getProductsByIdFunction = new lambda.Function(
+      this,
+      "getProductsByIdFunction",
+      {
+        runtime: lambda.Runtime.NODEJS_18_X,
+        handler: "getProductsById.handler",
+        code: lambda.Code.fromAsset("lambda"),
+      }
+    );
+
+    // API Gateway for getProductsList
+    const api = new apigateway.RestApi(this, "ProductServiceApi", {
+      restApiName: "Product Service",
+      defaultCorsPreflightOptions: {
+        allowOrigins: apigateway.Cors.ALL_ORIGINS,
+        allowMethods: apigateway.Cors.ALL_METHODS,
+      },
+    });
+
+    const products = api.root.addResource("products");
+    products.addMethod(
+      "GET",
+      new apigateway.LambdaIntegration(getProductsListFunction, {
+        proxy: false,
+        integrationResponses: [
+          {
+            statusCode: "200",
+            responseParameters: {
+              "method.response.header.Access-Control-Allow-Origin": "'*'",
+            },
+          },
+        ],
+      }),
+      {
+        methodResponses: [
+          {
+            statusCode: "200",
+            responseParameters: {
+              "method.response.header.Access-Control-Allow-Origin": true,
+            },
+          },
+        ],
+      }
+    );
+
+    const product = products.addResource("{productId}");
+    product.addMethod(
+      "GET",
+      new apigateway.LambdaIntegration(getProductsByIdFunction, {
+        proxy: true,
+        integrationResponses: [
+          {
+            statusCode: "200",
+            responseParameters: {
+              "method.response.header.Access-Control-Allow-Origin": "'*'",
+            },
+          },
+        ],
+      }),
+      {
+        methodResponses: [
+          {
+            statusCode: "200",
+            responseParameters: {
+              "method.response.header.Access-Control-Allow-Origin": true,
```

Comment:
you specify `"Access-Control-Allow-Origin": "'*'"` header incorrectly.

✅  it's working for `GET /products`, and I assume it's because of the code above
```ts
defaultCorsPreflightOptions: {
        allowOrigins: apigateway.Cors.ALL_ORIGINS
```

![image](https://github.com/deminiro/nodejs-aws-shop-react/assets/3858548/7a03937b-ddee-4994-bd23-03464527a654)

🔴 but it's not working for the `GET /products/:id`.

![image](https://github.com/deminiro/nodejs-aws-shop-react/assets/3858548/7d735f62-706b-41d3-b77d-146edd7cfbe6)

❗ this looks like a bug. Because of this I get CORS error when I call your API from other domains, such as editor.swagger.io

![image](https://github.com/deminiro/nodejs-aws-shop-react/assets/3858548/b4a7baf8-aaed-42c9-899c-b78d6bce628c)